base/fhs-validator.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+ROOTDIR="${1:-/rootfs}"
+
+DENYLIST_PATHS="lib64 lib bin sbin usr/lib64 usr/sbin"
+
+# make sure there's are no symlinks from DENYLIST_PATHS
+for DENYLIST_PATH in ${DENYLIST_PATHS}; do
+    # first check if the the file exists
+    if [[ -e "${ROOTDIR}/${DENYLIST_PATH}" ]]; then
+        if [[ -L "${ROOTDIR}/${DENYLIST_PATH}" ]]; then
+            echo "Found symlink ${ROOTDIR}/${DENYLIST_PATH} which is not allowed"
+            exit 1
+        fi
+
+        if [[ -d "${ROOTDIR}/${DENYLIST_PATH}" ]]; then
+            echo "Found directory ${ROOTDIR}/${DENYLIST_PATH} which is not allowed"
+            exit 1
+        fi
+    fi
+done
+
+# Test for extra files/directories
+ALLOWED_DIRS="usr etc bin sbin lib lib64 dev proc sys opt run var root tmp home"
```

Comment:
Hmm, actually, here we first check if there's something denied, and then deny everything not allowed. Perhaps a better and clearer approach would be to check for allowlist (not including non-usrmerged /bin ...), and then do an allowlist check on the usr itself (must include nothing more than bin, lib, share, etc, maybe sth else). This test shouldn't be ran on FHS itself since it creates /bin and others as symlinks